### PR TITLE
Support HTTP proxy in the opentok HTTP requests

### PR DESCRIPTION
**Pull request by juandebravo**
It could happen that you need to go through an HTTP Proxy to get access to opentok endpoint.

This PR makes feasible to define an HTTP proxy while creating an Opentok client. This value will be used in every HTTP request to the opentok endpoint.

### CODE REVIEW

File: opentok/opentok.py [hunk 1]
```diff
@@ -56,10 +56,20 @@ class OpenTok(object):
...
 
-    def __init__(self, api_key, api_secret, api_url='https://api.opentok.com'):
+    def __build_proxy_value(self, proxy):
+        """
+        Creates an dictionary with one key, `https`, and value the formatted proxy URL
+        """
+        if proxy.startswith('https://'):
+            return {'https': proxy}
+        else:
+            return {'https': 'https://' + proxy}
```
Comment (ggarber) [inline opentok/opentok.py]: What if the proxy is not an https proxy?

**Comment by aoberoi** [comment]
thanks for the PR @juandebravo. i agree that this configuration is useful. i have 2 comments:

1.  as @ggarber has noted, this API does not allow for defining both https and http (or any other protocol) proxies.

2.  i don't see a benefit in adding onto the opentok object constructor, if we could simply allow setting of the proxy using a setter on the opentok object. @ggarber has pointed out this would expose the configuration format of the underlying request library and make it more tightly coupled, but IMHO thats as good a format as any, and extensible enough that it could be abstracted later if the coupling becomes an issue.

how do you feel about refactoring to use a setter for the proxies?

**Comment by juandebravo** [comment]
Hi @aoberoi @ggarber ! thanks for the quick response.

Based on http://www.tokbox.com/blog/the-impact-of-googles-new-chrome-security-policy-on-webrtc/ and security topics, I think we should only accept `https` communications. If you think we might need `http` at any point, I'm fine addint it.

I'll work on the `setter` during today.

thanks.

**Comment by juandebravo** [comment]
Added the setter. I've removed the parameter in the constructor so now we're aligned with the requests expected format.

**Comment by juandebravo** [comment]
BTW I have no clue why the tests are failing :8ball:

**Comment by ggarber** [comment]
Apparently httpretty doesn't officially support python 3 and it had a bug:
https://github.com/gabrielfalcao/HTTPretty/issues/275

The fix has been merged in development but there is no new version released including that patch.

@aoberoi IMO we should merge this even if that test fails (it will pass in some days/week when they release httpretty) and maybe consider not using httpretty.

**Comment by juandebravo** [comment]
Thanks @ggarber 

@aoberoi appreciate if you canmerge and create a minor release with this functionality.

thanks!

**Comment by aoberoi** [comment]
I was considering the value of doing some validation on the proxy config value in the setter. But now I actually think its fine to let errors during an actual request propagate back to the caller instead of doing any validation upfront. I'll cut the release now. Thanks!!

**Pull request merged**